[drape] minVisibleScale-related cleanup and optimization

pastk opened the pull request:
1. Removed minVisibleScale param passing which is not used anymore (after #4645).
2. Removed overlays masks handling which is not used anymore (after 3e3d2d36a483beac319a2028fa5cbda79d672bb2).
3. Optimize determining of main overlays type by storing maxOverlaysPriority once per classificator type instead of doing it for every multi-type feature.

4. Try to re-enable some discard render bucket optimization. More on this later.
Upd: moved it from this PR to https://github.com/organicmaps/organicmaps/pull/5921

pastk commented:
There is some render bucket discard optimization, which is not working on master.
At first I thought I accidentally broke it in 0ac765a27f3825fba7e614495da736f01c5ed1a6, but actually looks like it didn't work before already (here is a code with debug output which is supposed to work, but it doesn't: https://github.com/organicmaps/organicmaps/tree/pastk-drape-bucket-discard-TEST).

It looks like passing minVisibleScale param by reference into `insertShape` doesn't work.

Originially the optimization was added in https://github.com/mapsme/omim/commit/519e60dc3f675035281acf8fee4b2203a40e19fa and then modified in https://github.com/mapsme/omim/pull/2570 (basically it was enabled for point features and areas with point styles only, i.e. overlays only, but w/o lines pathtexts/shields).

@vng you were reviewing the changes back then so maybe you remember the logic behind it?

I don't really understand the surrounding code and how this optimization is supposed to work:
- why there are several render buckets per render group?
- why render buckets are not discarded immediately upon zoom level change?
- why this optimization was (seemingly) used for overlays only?

The current code in this PR (commit 2) actually works and some buckets are being discarded, though I don't see any difference on the screen (not the picture/animation, not the speed) and its risky to re-enable it without fully understanding how it works.

vng commented on drape/overlay_handle.cpp:
```diff
@@ -228,26 +228,19 @@ std::string SquareHandle::GetOverlayDebugInfo()
+  return (static_cast<uint64_t>(depth) << 24) |
          (static_cast<uint64_t>(rank) << 16) |
-         static_cast<uint64_t>(0xFFFF);
+         static_cast<uint64_t>(0xFF0000000000FFFF);
```
Is there any reason to skip low 2 bytes? Are they assigned later?

pastk replied:
They're used for special priority modes.

pastk replied:
Changed it to const.

vng commented on drape_frontend/rule_drawer.cpp:
```diff
@@ -492,29 +487,29 @@ void RuleDrawer::operator()(FeatureType & f)
-  /// @todo Remove passing of minVisibleScale arg everywhere.
-  int minVisibleScale = 0;
-  auto insertShape = [this, &minVisibleScale](drape_ptr<MapShape> && shape)
+  int const minOverlaysZoom = s.m_minOverlaysZoom;
```
Ok, if s.m_minOverlaysZoom == 0 ?

pastk replied:
ATM on master its 0 always.

pastk commented:
> There is some render bucket discard optimization, which is not working on master.

I removed this optimization re-enabling commit from this PR as I'm not certain about it. Just left a comment and will create a new PR for it.

Added removal of unused (after 3e3d2d36a483beac319a2028fa5cbda79d672bb2) overlay masks handling.

PTAL!

pastk commented:
> I removed this optimization re-enabling commit from this PR as I'm not certain about it. Just left a comment and will create a new PR for it.

A new PR:
- https://github.com/organicmaps/organicmaps/pull/5921
